src/components/subscriptions/hooks/licenseManagerHooks.js

```diff
@@ -0,0 +1,208 @@
+    isLoading,
+    error,
+  };
+};
+
+export const useSubscriptionUsers = ({ subscriptionUUID, options: { status, search, page } }) => {
+  const [subscriptionUsers, setSubscriptionUsers] = useState({
+    results: [],
+    count: 0,
+    next: null,
+    previous: null,
+  });
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  const fetch = () => {
+    setIsLoading(true);
+
+    if (!subscriptionUUID) {
+      return;
+    }
+
+    LicenseManagerApiService.fetchSubscriptionUsers(subscriptionUUID, { status, search, page })
+      .then((response) => {
+        setIsLoading(false);
+        setSubscriptionUsers(response.data);
+      })
+      .catch((err) => {
+        NewRelicService.logAPIErrorResponse(err);
+        setIsLoading(false);
+        setError(err);
+      });
+  };
+
+  return {
+    fetch,
+    subscriptionUsers,
+    isLoading,
+    error,
+  };
+};
+
+/*
+This is a helper hook that provides all the subscription data in a single place.
+*/
+export const useSubscriptionData = ({ status, search, page = 1 }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [errors, setErrors] = useState(null);
+  const [data, setData] = useState({
+    subscriptions: {},
+    overview: {},
+    subscriptionUsers: {},
+  });
+
+  const {
+    fetch: fetchSubscriptions,
+    subscriptions,
+    isLoading: isLoadingSubscriptions,
+    error: subscriptionsError,
+  } = useSubscriptions();
+  const {
+    fetch: fetchSubscriptionUsersOverview,
+    subscriptionUsersOverview,
+    isLoading: isLoadingSubscriptionUsersOverview,
+    error: subscriptionUsersOverviewError,
+  } = useSubscriptionUsersOverview({ subscriptionUUID: subscriptions?.results[0]?.uuid });
+
+  const {
+    fetch: fetchSubscriptionUsers,
+    subscriptionUsers,
+    isLoading: isLoadingSubscriptionUsers,
+    error: subscriptionUsersError,
+  } = useSubscriptionUsers({
+    subscriptionUUID: subscriptions?.results[0]?.uuid,
+    options: { status, search, page },
+  });
+
+  const fetch = () => {
+    // This method will fetch subscriptions and after the successfully fetching subscriptions
+    // data, it will trigger the fetch operation of related data.
+    setIsLoading(true);
+    fetchSubscriptions();
+  };
+
+  useEffect(() => fetchSubscriptionUsersOverview(), [subscriptions]);
+  useEffect(
+    () => fetchSubscriptionUsers(),
+    [subscriptions, status, search, page],
+  );
+
+  useEffect(() => {
+    if (
+      isLoadingSubscriptions || isLoadingSubscriptionUsersOverview || isLoadingSubscriptionUsers
+    ) {
+      // Keep the status as isLoading until all requests complete.
+      // setIsLoading(true);
+    } else if (subscriptionsError || subscriptionUsersOverviewError || subscriptionUsersError) {
+      setIsLoading(false);
+      setErrors(prevState => ({
+        ...(prevState || {}),
+        ...(subscriptionsError ? { [SUBSCRIPTIONS]: NETWORK_ERROR_MESSAGE } : {}),
```

Comment:
Would it make sense to construct the object that this function returns outside of the function, e.g.:

```
const updatedErrors = {};
updatedErrors[SUBSCRIPTIONS] = subscriptionsError && NETWORK_ERROR_MESSAGE;
updatedErrors[SUBSCRIPTION_USERS_OVERVIEW] = subscriptionUsersOverviewError && NETWORK_ERROR_MESSAGE;

setErrors(prevState => ({
  ...(prevState || {}),
  ...updatedErrors,
});
```

Reply:
`updatedErrors[SUBSCRIPTIONS] = subscriptionsError && NETWORK_ERROR_MESSAGE;` causes an issue as it adds an attribute with `null` as value. However, I have moved error object construction outside of the function and tried to clean it up a bit.